include/cuco/detail/static_set/kernels.cuh

```diff
@@ -89,6 +90,257 @@ CUCO_KERNEL void find(InputIt first, cuco::detail::index_type n, OutputIt output
+          typename InputIt,
+          typename OutputIt1,
+          typename OutputIt2,
+          typename AtomicT,
+          typename Ref>
+__device__ void group_retrieve(InputIt first,
+                               cuco::detail::index_type n,
+                               OutputIt1 output_probe,
+                               OutputIt2 output_match,
+                               AtomicT* counter,
+                               Ref ref)
+{
+  namespace cg = cooperative_groups;
+
+  using size_type = typename Ref::size_type;
+  using ProbeKey  = typename std::iterator_traits<InputIt>::value_type;
+  using Key       = typename Ref::key_type;
+
+  auto constexpr tile_size   = Ref::cg_size;
+  auto constexpr window_size = Ref::window_size;
+
+  auto idx          = cuco::detail::global_thread_id() / tile_size;
+  auto const stride = cuco::detail::grid_stride() / tile_size;
+  auto const block  = cg::this_thread_block();
+  auto const tile   = cg::tiled_partition<tile_size>(block);
+
+  auto constexpr flushing_tile_size = cuco::detail::warp_size() / window_size;
+  // random choice to tune
+  auto constexpr flushing_buffer_size = 2 * flushing_tile_size;
+  auto constexpr num_flushing_tiles   = BlockSize / flushing_tile_size;
+  auto constexpr max_matches          = flushing_tile_size / tile_size;
+
+  auto const flushing_tile    = cg::tiled_partition<flushing_tile_size>(block);
+  auto const flushing_tile_id = block.thread_rank() / flushing_tile_size;
+
+  __shared__ cuco::pair<ProbeKey, Key> flushing_tile_buffer[num_flushing_tiles][flushing_tile_size];
+  // per flushing-tile counter to track number of filled elements
+  __shared__ cuda::atomic<size_type, cuda::thread_scope_block> flushing_counter[num_flushing_tiles];
+
+#if defined(CUCO_HAS_CG_INVOKE_ONE)
+  cg::invoke_one(flushing_tile, [&]() { flushing_counter[flushing_tile_id] = 0; });
+#else
+  if (flushing_tile.thread_rank() == 0) { flushing_counter[flushing_tile_id] = 0; }
+#endif
+  flushing_tile.sync();  // sync still needed since cg.any doesn't imply a memory barrier
+
+  while (flushing_tile.any(idx < n)) {
+    bool active_flag = idx < n;
+    auto const active_flushing_tile =
+      cg::binary_partition<flushing_tile_size>(flushing_tile, active_flag);
+    if (active_flag) {
+      auto const found = ref.find(tile, *(first + idx));
+#if defined(CUCO_HAS_CG_INVOKE_ONE)
+      if (found != ref.end()) {
+        cg::invoke_one(tile, [&]() {
+          auto const offset =
+            flushing_counter[flushing_tile_id].fetch_add(1, cuda::std::memory_order_relaxed);
+          flushing_tile_buffer[flushing_tile_id][offset] = cuco::pair{*(first + idx), *found};
+        });
+      }
+#else
+      if (tile.thread_rank() == 0 and found != ref.end()) {
+        auto const offset =
+          flushing_counter[flushing_tile_id].fetch_add(1, cuda::std::memory_order_relaxed);
+        flushing_tile_buffer[flushing_tile_id][offset] = cuco::pair{*(first + idx), *found};
```

Comment:
```suggestion
        flushing_tile_buffer[flushing_tile_id][offset] = {*(first + idx), *found};
```

Just a suggestion/my preference. Some find stating the type explicitly more readable.